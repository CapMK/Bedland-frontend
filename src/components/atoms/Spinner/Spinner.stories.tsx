import type { Meta, StoryObj } from '@storybook/react';
import GlobalStyle from 'theme/globalStyles';
import ThemeProvider from 'theme/ThemeContext';
import Spinner from './Spinner';

const meta = {
  title: 'Atoms/Spinner',
  component: Spinner,
  tags: ['autodocs'],
  decorators: [
    (Story) => (
      <ThemeProvider>
        <GlobalStyle />
        {Story()}
      </ThemeProvider>
    ),
  ],
} satisfies Meta<typeof Spinner>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    size: 24,
  },
};
