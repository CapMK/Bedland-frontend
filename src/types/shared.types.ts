export type Getters<T extends string, K> = {
  [k in T]: K;
};

export type Building = {
  id: string;
  name: string;
  address: string;
  managerId: string;
  floors: string;
  createDate: string;
  updateDate: string;
  version: number;
};

export type Flat = {
  id: string;
  version: number;
  createDate: string;
  updateDate: string;
  buildingId: string;
  number: string;
  floor: string;
  shapePath: string;
};

export type FlatWithAddress = Flat & { address: string; city: string };

export type ResidentsFlats = {
  flatId: string;
  flatNumber: string;
  flatAddress: string;
  flatCreateDate: string;
};

export type BuildingWithCity = Building & { city: string };

export type ResidentsFlatsWithCity = ResidentsFlats & { city: string };
