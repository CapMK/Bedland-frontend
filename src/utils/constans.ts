export const HEADER_HEIGHT = 70;
export const FOOTER_HEIGHT = 52;
export const LOGIN_INPUT_WIDTH = 19.875;

// TODO
// BASE_URL points to json server
// Replace with actual API url when BE is ready
export const BASE_URL = 'http://localhost:3001';
