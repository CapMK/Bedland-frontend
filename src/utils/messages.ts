const ERROR_MESSAGES = {
  resetPasswordPage: {
    notWithPasswordsRules:
      'Password is incorrect, please try again after reviewing the rules above',

    passwordsDoesNotMatchUp: 'Passwords do not match to each other',
  },
  loginPage: {
    errorMessage: 'Your login or password is incorrect.',
  },
};

export default ERROR_MESSAGES;
