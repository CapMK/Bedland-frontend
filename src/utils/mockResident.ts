const mockResidentList = [
  {
    id: 1,
    version: 0,
    createDate: '2023-02-14T00:00:00',
    updateDate: '2023-02-14T00:00:00',
    login: 'iSutherland',
    password: 'IsABEl',
    name: 'Isabel',
    lastName: 'Sutherland',
    email: 'isabel.sutherland@email.com',
    phoneNumber: '123456789',
    flatId: 1,
    owner: true,
  },
  {
    id: 2,
    version: 0,
    createDate: '2023-02-14T00:00:00',
    updateDate: '2023-02-14T00:00:00',
    login: 'clWeiss',
    password: 'ClWeiss',
    name: 'Claude',
    lastName: 'Weiss',
    email: 'Claude.Weiss@email.com',
    phoneNumber: '123456789',
    flatId: 2,
    owner: false,
  },
];

export default mockResidentList;
