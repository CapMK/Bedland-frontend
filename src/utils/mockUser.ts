import { User } from 'store/reducers/user/user_slice';

export const mockUserCredentials = {
  residentUsername: 'testmember',
  managerUsername: 'testmanager',
  password: '!Test123!',
};

export const loggedInManagerUser = {
  loggedIn: true,
  status: 'success',
  error: undefined,
  userId: '122',
  login: mockUserCredentials.managerUsername,
  token: undefined,
  role: 'manager',
} as User;

export const loggedInResidentUser = {
  loggedIn: true,
  status: 'success',
  error: undefined,
  userId: '122',
  login: mockUserCredentials.residentUsername,
  token: undefined,
  role: 'resident',
} as User;

export const loggedOutManagerUser = {
  loggedIn: false,
  status: 'idle',
  userId: undefined,
  login: undefined,
  token: undefined,
  role: 'manager',
} as User;

export const loggedOutResidentUser = {
  loggedIn: false,
  status: 'idle',
  userId: undefined,
  login: undefined,
  token: undefined,
  role: 'resident',
} as User;
