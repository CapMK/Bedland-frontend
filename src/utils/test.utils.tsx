import { PropsWithChildren } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { configureStore, PreloadedState } from '@reduxjs/toolkit';
import { AppStore, rootMiddleware, rootReducer, RootState } from 'store/store';
import { Provider } from 'react-redux';
import { MemoryRouter, useLocation } from 'react-router-dom';
import ThemeProvider from 'theme/ThemeContext';

const LocationDisplay = () => {
  const location = useLocation();
  return <div data-testid='location-display'>{location.pathname}</div>;
};

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>;
  initialUrl?: string;
  store?: AppStore;
}

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    initialUrl = '/',
    store = configureStore({
      reducer: rootReducer,
      middleware: rootMiddleware,
      preloadedState,
    }),
    ...renderOptions
  }: ExtendedRenderOptions = {},
) {
  const Wrapper = ({ children }: PropsWithChildren<object>): JSX.Element => (
    <ThemeProvider>
      <MemoryRouter initialEntries={[initialUrl]}>
        <Provider store={store}>{children}</Provider>
        <LocationDisplay />
      </MemoryRouter>
    </ThemeProvider>
  );

  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export * from '@testing-library/react';
