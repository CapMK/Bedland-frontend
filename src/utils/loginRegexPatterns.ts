import { InputVariant, TRegister } from 'components/atoms/Input/Input.types';
import { Getters } from 'types/shared.types';

type LoginRegexProps = Getters<InputVariant, TRegister>;

const loginRegexPattern: LoginRegexProps = {
  login: {
    pattern: /^[a-zA-Z0-9]+$/,
    required: true,
    maxLength: 36,
    minLength: 8,
  },
  password: {
    pattern: /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#$%^&*(),<.>/?;:'"]).{8,24}$/,
    required: true,
    maxLength: 36,
    minLength: 8,
  },
  confirmPassword: {
    required: true,
  },
};

export default loginRegexPattern;
// ^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$ <- regex for password that should work.
// /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/,
