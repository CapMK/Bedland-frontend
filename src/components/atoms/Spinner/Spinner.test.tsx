import { renderWithProviders } from 'utils/test.utils';
import Spinner from './Spinner';
import { SpinnerProps } from './Spinner.types';

describe('Scroll', () => {
  const item = ({ size }: SpinnerProps) => renderWithProviders(<Spinner size={size} />, {});
  it('should renders correctly Spinner', () => {
    expect(
      item({
        size: 24,
      }),
    ).toMatchSnapshot();
  });
});
