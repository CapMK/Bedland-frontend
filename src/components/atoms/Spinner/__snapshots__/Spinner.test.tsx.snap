// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Scroll > should renders correctly Spinner 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <svg
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <circle
            cx="12"
            cy="3"
            r="1"
          >
            <animate
              attributeName="r"
              begin="0;spinner_tKsu.end-0.5s"
              calcMode="spline"
              dur="0.6s"
              id="spinner_7Z73"
              keySplines=".27,.42,.37,.99;.53,0,.61,.73"
              values="1;2;1"
            />
          </circle>
          <circle
            cx="16.50"
            cy="4.21"
            r="1"
          >
            <animate
              attributeName="r"
              begin="spinner_7Z73.begin+0.1s"
              calcMode="spline"
              dur="0.6s"
              id="spinner_Wd87"
              keySplines=".27,.42,.37,.99;.53,0,.61,.73"
              values="1;2;1"
            />
          </circle>
          <circle
            cx="7.50"
            cy="4.21"
            r="1"
          >
            <animate
              attributeName="r"
              begin="spinner_9Qlc.begin+0.1s"
              calcMode="spline"
              dur="0.6s"
              id="spinner_tKsu"
              keySplines=".27,.42,.37,.99;.53,0,.61,.73"
              values="1;2;1"
            />
          </circle>
          <circle
            cx="19.79"
            cy="7.50"
            r="1"
          >
            <animate
              attributeName="r"
              begin="spinner_Wd87.begin+0.1s"
              calcMode="spline"
              dur="0.6s"
              id="spinner_lMMO"
              keySplines=".27,.42,.37,.99;.53,0,.61,.73"
              values="1;2;1"
            />
          </circle>
          <circle
            cx="4.21"
            cy="7.50"
            r="1"
          >
            <animate
              attributeName="r"
              begin="spinner_Khxv.begin+0.1s"
              calcMode="spline"
              dur="0.6s"
              id="spinner_9Qlc"
              keySplines=".27,.42,.37,.99;.53,0,.61,.73"
              values="1;2;1"
            />
          </circle>
          <circle
            cx="21.00"
            cy="12.00"
            r="1"
          >
            <animate
              attributeName="r"
              begin="spinner_lMMO.begin+0.1s"
              calcMode="spline"
              dur="0.6s"
              id="spinner_5L9t"
              keySplines=".27,.42,.37,.99;.53,0,.61,.73"
              values="1;2;1"
            />
          </circle>
          <circle
            cx="3.00"
            cy="12.00"
            r="1"
          >
            <animate
              attributeName="r"
              begin="spinner_ld6P.begin+0.1s"
              calcMode="spline"
              dur="0.6s"
              id="spinner_Khxv"
              keySplines=".27,.42,.37,.99;.53,0,.61,.73"
              values="1;2;1"
            />
          </circle>
          <circle
            cx="19.79"
            cy="16.50"
            r="1"
          >
            <animate
              attributeName="r"
              begin="spinner_5L9t.begin+0.1s"
              calcMode="spline"
              dur="0.6s"
              id="spinner_BfTD"
              keySplines=".27,.42,.37,.99;.53,0,.61,.73"
              values="1;2;1"
            />
          </circle>
          <circle
            cx="4.21"
            cy="16.50"
            r="1"
          >
            <animate
              attributeName="r"
              begin="spinner_XyBs.begin+0.1s"
              calcMode="spline"
              dur="0.6s"
              id="spinner_ld6P"
              keySplines=".27,.42,.37,.99;.53,0,.61,.73"
              values="1;2;1"
            />
          </circle>
          <circle
            cx="16.50"
            cy="19.79"
            r="1"
          >
            <animate
              attributeName="r"
              begin="spinner_BfTD.begin+0.1s"
              calcMode="spline"
              dur="0.6s"
              id="spinner_7gAK"
              keySplines=".27,.42,.37,.99;.53,0,.61,.73"
              values="1;2;1"
            />
          </circle>
          <circle
            cx="7.50"
            cy="19.79"
            r="1"
          >
            <animate
              attributeName="r"
              begin="spinner_HiSl.begin+0.1s"
              calcMode="spline"
              dur="0.6s"
              id="spinner_XyBs"
              keySplines=".27,.42,.37,.99;.53,0,.61,.73"
              values="1;2;1"
            />
          </circle>
          <circle
            cx="12"
            cy="21"
            r="1"
          >
            <animate
              attributeName="r"
              begin="spinner_7gAK.begin+0.1s"
              calcMode="spline"
              dur="0.6s"
              id="spinner_HiSl"
              keySplines=".27,.42,.37,.99;.53,0,.61,.73"
              values="1;2;1"
            />
          </circle>
          <animatetransform
            attributeName="transform"
            dur="6s"
            repeatCount="indefinite"
            type="rotate"
            values="360 12 12;0 12 12"
          />
        </g>
      </svg>
      <div
        data-testid="location-display"
      >
        /
      </div>
    </div>
  </body>,
  "container": <div>
    <svg
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g>
        <circle
          cx="12"
          cy="3"
          r="1"
        >
          <animate
            attributeName="r"
            begin="0;spinner_tKsu.end-0.5s"
            calcMode="spline"
            dur="0.6s"
            id="spinner_7Z73"
            keySplines=".27,.42,.37,.99;.53,0,.61,.73"
            values="1;2;1"
          />
        </circle>
        <circle
          cx="16.50"
          cy="4.21"
          r="1"
        >
          <animate
            attributeName="r"
            begin="spinner_7Z73.begin+0.1s"
            calcMode="spline"
            dur="0.6s"
            id="spinner_Wd87"
            keySplines=".27,.42,.37,.99;.53,0,.61,.73"
            values="1;2;1"
          />
        </circle>
        <circle
          cx="7.50"
          cy="4.21"
          r="1"
        >
          <animate
            attributeName="r"
            begin="spinner_9Qlc.begin+0.1s"
            calcMode="spline"
            dur="0.6s"
            id="spinner_tKsu"
            keySplines=".27,.42,.37,.99;.53,0,.61,.73"
            values="1;2;1"
          />
        </circle>
        <circle
          cx="19.79"
          cy="7.50"
          r="1"
        >
          <animate
            attributeName="r"
            begin="spinner_Wd87.begin+0.1s"
            calcMode="spline"
            dur="0.6s"
            id="spinner_lMMO"
            keySplines=".27,.42,.37,.99;.53,0,.61,.73"
            values="1;2;1"
          />
        </circle>
        <circle
          cx="4.21"
          cy="7.50"
          r="1"
        >
          <animate
            attributeName="r"
            begin="spinner_Khxv.begin+0.1s"
            calcMode="spline"
            dur="0.6s"
            id="spinner_9Qlc"
            keySplines=".27,.42,.37,.99;.53,0,.61,.73"
            values="1;2;1"
          />
        </circle>
        <circle
          cx="21.00"
          cy="12.00"
          r="1"
        >
          <animate
            attributeName="r"
            begin="spinner_lMMO.begin+0.1s"
            calcMode="spline"
            dur="0.6s"
            id="spinner_5L9t"
            keySplines=".27,.42,.37,.99;.53,0,.61,.73"
            values="1;2;1"
          />
        </circle>
        <circle
          cx="3.00"
          cy="12.00"
          r="1"
        >
          <animate
            attributeName="r"
            begin="spinner_ld6P.begin+0.1s"
            calcMode="spline"
            dur="0.6s"
            id="spinner_Khxv"
            keySplines=".27,.42,.37,.99;.53,0,.61,.73"
            values="1;2;1"
          />
        </circle>
        <circle
          cx="19.79"
          cy="16.50"
          r="1"
        >
          <animate
            attributeName="r"
            begin="spinner_5L9t.begin+0.1s"
            calcMode="spline"
            dur="0.6s"
            id="spinner_BfTD"
            keySplines=".27,.42,.37,.99;.53,0,.61,.73"
            values="1;2;1"
          />
        </circle>
        <circle
          cx="4.21"
          cy="16.50"
          r="1"
        >
          <animate
            attributeName="r"
            begin="spinner_XyBs.begin+0.1s"
            calcMode="spline"
            dur="0.6s"
            id="spinner_ld6P"
            keySplines=".27,.42,.37,.99;.53,0,.61,.73"
            values="1;2;1"
          />
        </circle>
        <circle
          cx="16.50"
          cy="19.79"
          r="1"
        >
          <animate
            attributeName="r"
            begin="spinner_BfTD.begin+0.1s"
            calcMode="spline"
            dur="0.6s"
            id="spinner_7gAK"
            keySplines=".27,.42,.37,.99;.53,0,.61,.73"
            values="1;2;1"
          />
        </circle>
        <circle
          cx="7.50"
          cy="19.79"
          r="1"
        >
          <animate
            attributeName="r"
            begin="spinner_HiSl.begin+0.1s"
            calcMode="spline"
            dur="0.6s"
            id="spinner_XyBs"
            keySplines=".27,.42,.37,.99;.53,0,.61,.73"
            values="1;2;1"
          />
        </circle>
        <circle
          cx="12"
          cy="21"
          r="1"
        >
          <animate
            attributeName="r"
            begin="spinner_7gAK.begin+0.1s"
            calcMode="spline"
            dur="0.6s"
            id="spinner_HiSl"
            keySplines=".27,.42,.37,.99;.53,0,.61,.73"
            values="1;2;1"
          />
        </circle>
        <animatetransform
          attributeName="transform"
          dur="6s"
          repeatCount="indefinite"
          type="rotate"
          values="360 12 12;0 12 12"
        />
      </g>
    </svg>
    <div
      data-testid="location-display"
    >
      /
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
